models/userModel.ts

```diff
@@ -67,6 +73,7 @@ export declare namespace UserMethods {
     export type GetUserById = (id: string) => Promise<User | null>;
     // Existing prisma generated user method types
     export type FindUnique = typeof userModel.findUnique;
+    export type DeleteUser = (id: string | undefined) => Promise<User | null>;
```

Comment:
For the type I would suggest not allowing undefined as an input as it would lead to a 500 Internal Server Error. It would be better to make sure there is a value before calling the method.

prismaUser.delete returns an exception if the user doesn't exist: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#delete

I got around this by first checking if the user exists and returning null if not: 
```
updatePost: async (postId, postData) => {
        const post = await prismaPost.findUnique({
            where: {
                id: postId,
            },
        });

        if (!post) return null;

        const tagIds = getTagIds(postData.postTags);

        return prismaPost.update({
            where: {
                id: postId,
            },
            data: {
                ...postData,
                postTags: {
                    deleteMany: {
                        postId: postId,
                    },
                    createMany: {
                        data: await tagIds,
                    },
                },
            },
            include: {
                postTags: {
                    include: {
                        tag: true,
                    },
                },
                comments: true,
                likes: true,
            },
        });
    }
```

Another option is to run the check in the controller. I did this for deletePost but will probably change it to the above method so this login remains in the data layer.

```
// Check if the post exists
        const post = await DBGetPostById(postId);
        if (!post)
            return res
                .status(400)
                .json({ status: "error", message: "Post not found" });
```